{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e827ecaf-ae57-46a5-a167-addf1a45d75b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(readr)\n",
    "library(dplyr)\n",
    "library(stringr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "43db16d4-a8b8-42ad-b37c-28001b50aa83",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/mnt/l/Sia_git/metagen_files/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15346a30-9dbb-468a-adf2-311d40ca7e0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#setwd to metagen_files directory PATH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "51cae074-e52c-46b6-994c-c25769847bb7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `region)` -> `region)...50`\n",
      "\u001b[36m•\u001b[39m `region)` -> `region)...63`\n",
      "Warning message:\n",
      "“\u001b[1m\u001b[22mOne or more parsing issues, call `problems()` on your data frame for details, e.g.:\n",
      "  dat <- vroom(...)\n",
      "  problems(dat)”\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m1290\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m69\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m  (39): Run, Assay Type, BioProject, BioSample, Center Name, Collection_D...\n",
      "\u001b[32mdbl\u001b[39m   (8): AGE, Bytes, geographic_location_(latitude), geographic_location_(...\n",
      "\u001b[33mlgl\u001b[39m  (14): body_product, AJCC_Stage, alias, ENA_checklist, environmental_pac...\n",
      "\u001b[34mdttm\u001b[39m  (4): INSDC_first_public, INSDC_last_update, ReleaseDate, create_date\n",
      "\u001b[34mdate\u001b[39m  (4): ENA-FIRST-PUBLIC (run), ENA-LAST-UPDATE (run), ENA_first_public, ...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 69</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>AGE</th><th scope=col>Assay Type</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>Bytes</th><th scope=col>Center Name</th><th scope=col>Collection_Date</th><th scope=col>Consent</th><th scope=col>DATASTORE filetype</th><th scope=col>⋯</th><th scope=col>environmental_package</th><th scope=col>gender</th><th scope=col>geographic_location_(country_and/or_sea</th><th scope=col>region)...63</th><th scope=col>localization</th><th scope=col>SRA_accession</th><th scope=col>tissue_type</th><th scope=col>TITLE</th><th scope=col>TNM_Stage</th><th scope=col>BMI</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ERR478965</td><td>35</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2466965</td><td>1291630481</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR478979</td><td>61</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2467031</td><td> 757288174</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR478980</td><td>65</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2467021</td><td> 725101549</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR478983</td><td>65</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2467021</td><td> 721823859</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR478990</td><td>77</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2467036</td><td> 922844007</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR478991</td><td>77</td><td>WGS</td><td>PRJEB6070</td><td>SAMEA2467036</td><td> 918721357</td><td>EMBL - GERMANY</td><td>2004/2006</td><td>public</td><td>fastq,run.zq,sra</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 69\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & AGE & Assay Type & BioProject & BioSample & Bytes & Center Name & Collection\\_Date & Consent & DATASTORE filetype & ⋯ & environmental\\_package & gender & geographic\\_location\\_(country\\_and/or\\_sea & region)...63 & localization & SRA\\_accession & tissue\\_type & TITLE & TNM\\_Stage & BMI\\\\\n",
       " <chr> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t ERR478965 & 35 & WGS & PRJEB6070 & SAMEA2466965 & 1291630481 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t ERR478979 & 61 & WGS & PRJEB6070 & SAMEA2467031 &  757288174 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t ERR478980 & 65 & WGS & PRJEB6070 & SAMEA2467021 &  725101549 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t ERR478983 & 65 & WGS & PRJEB6070 & SAMEA2467021 &  721823859 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t ERR478990 & 77 & WGS & PRJEB6070 & SAMEA2467036 &  922844007 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t ERR478991 & 77 & WGS & PRJEB6070 & SAMEA2467036 &  918721357 & EMBL - GERMANY & 2004/2006 & public & fastq,run.zq,sra & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 69\n",
       "\n",
       "| Run &lt;chr&gt; | AGE &lt;dbl&gt; | Assay Type &lt;chr&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | Bytes &lt;dbl&gt; | Center Name &lt;chr&gt; | Collection_Date &lt;chr&gt; | Consent &lt;chr&gt; | DATASTORE filetype &lt;chr&gt; | ⋯ ⋯ | environmental_package &lt;lgl&gt; | gender &lt;lgl&gt; | geographic_location_(country_and/or_sea &lt;lgl&gt; | region)...63 &lt;lgl&gt; | localization &lt;lgl&gt; | SRA_accession &lt;lgl&gt; | tissue_type &lt;lgl&gt; | TITLE &lt;lgl&gt; | TNM_Stage &lt;lgl&gt; | BMI &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ERR478965 | 35 | WGS | PRJEB6070 | SAMEA2466965 | 1291630481 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| ERR478979 | 61 | WGS | PRJEB6070 | SAMEA2467031 |  757288174 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| ERR478980 | 65 | WGS | PRJEB6070 | SAMEA2467021 |  725101549 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| ERR478983 | 65 | WGS | PRJEB6070 | SAMEA2467021 |  721823859 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| ERR478990 | 77 | WGS | PRJEB6070 | SAMEA2467036 |  922844007 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| ERR478991 | 77 | WGS | PRJEB6070 | SAMEA2467036 |  918721357 | EMBL - GERMANY | 2004/2006 | public | fastq,run.zq,sra | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       AGE Assay Type BioProject BioSample    Bytes      Center Name   \n",
       "1 ERR478965 35  WGS        PRJEB6070  SAMEA2466965 1291630481 EMBL - GERMANY\n",
       "2 ERR478979 61  WGS        PRJEB6070  SAMEA2467031  757288174 EMBL - GERMANY\n",
       "3 ERR478980 65  WGS        PRJEB6070  SAMEA2467021  725101549 EMBL - GERMANY\n",
       "4 ERR478983 65  WGS        PRJEB6070  SAMEA2467021  721823859 EMBL - GERMANY\n",
       "5 ERR478990 77  WGS        PRJEB6070  SAMEA2467036  922844007 EMBL - GERMANY\n",
       "6 ERR478991 77  WGS        PRJEB6070  SAMEA2467036  918721357 EMBL - GERMANY\n",
       "  Collection_Date Consent DATASTORE filetype ⋯ environmental_package gender\n",
       "1 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "2 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "3 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "4 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "5 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "6 2004/2006       public  fastq,run.zq,sra   ⋯ NA                    NA    \n",
       "  geographic_location_(country_and/or_sea region)...63 localization\n",
       "1 NA                                      NA           NA          \n",
       "2 NA                                      NA           NA          \n",
       "3 NA                                      NA           NA          \n",
       "4 NA                                      NA           NA          \n",
       "5 NA                                      NA           NA          \n",
       "6 NA                                      NA           NA          \n",
       "  SRA_accession tissue_type TITLE TNM_Stage BMI\n",
       "1 NA            NA          NA    NA        NA \n",
       "2 NA            NA          NA    NA        NA \n",
       "3 NA            NA          NA    NA        NA \n",
       "4 NA            NA          NA    NA        NA \n",
       "5 NA            NA          NA    NA        NA \n",
       "6 NA            NA          NA    NA        NA "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#colon cancer study\n",
    "\n",
    "#case 01 - France\n",
    "Study_1_metadata_France_colon_rectar <- read_csv(\"Metadata/Study_1_metadata_France_colon_rectar.csv\")\n",
    "head(Study_1_metadata_France_colon_rectar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fcaa02c-a280-43d3-a3a3-e82fc4e73c9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "statics <- as.data.frame(table(Study_1_metadata_France_colon_rectar$host_subject_id))\n",
    "\n",
    "colnames(statics)[colnames(statics) == \"Var1\" ] <- \"host_subject_id\"\n",
    "connected_meta <- inner_join(statics, Study_1_metadata_France_colon_rectar, by = \"host_subject_id\")\n",
    "\n",
    "#select specific collumns\n",
    "final_meta_study1 <- connected_meta[, c(\"host_subject_id\", \"Freq\", \"diagnosis\")]\n",
    "\n",
    "#get uniques lines\n",
    "final_meta_study_1_unique <- distinct(final_meta_study1, .keep_all = TRUE)\n",
    "\n",
    "#filter by country\n",
    "\n",
    "study1_france <- final_meta_study_1_unique %>%\n",
    "  filter(str_detect(host_subject_id, \"FR\"))\n",
    "\n",
    "#filter France by condition\n",
    "study1_france_cancer <- study1_france[study1_france$diagnosis == \"Cancer\", ]\n",
    "study1_france_small_adeno <-  study1_france[study1_france$diagnosis == \"Small adenoma\", ]\n",
    "study1_france_large_adeno <-  study1_france[study1_france$diagnosis == \"Large adenoma\", ]\n",
    "study1_france_normal <-  study1_france[study1_france$diagnosis == \"Normal\", ]\n",
    "\n",
    "#take ID from cancer - France\n",
    "france_cancer_ID <- inner_join(study1_france_cancer, Study_1_metadata_France_colon_rectar, by = \"host_subject_id\")\n",
    "#take collumns\n",
    "france_cancer_get_ID <- france_cancer_ID %>% select(Run)\n",
    "\n",
    "write_tsv(france_cancer_get_ID, \"france_cancer_get_ID.tsv\")\n",
    "\n",
    "#take ID from small adeno - France\n",
    "france_small_ID <- inner_join(study1_france_small_adeno, Study_1_metadata_France_colon_rectar, by = \"host_subject_id\")\n",
    "#take collumns\n",
    "france_small_get_ID <- france_small_ID %>% select(Run)\n",
    "\n",
    "write_tsv(france_small_get_ID, \"france_small_adenoma_get_ID.tsv\")\n",
    "\n",
    "#take ID from large adeno - France\n",
    "france_large_ID <- inner_join(study1_france_large_adeno, Study_1_metadata_France_colon_rectar, by = \"host_subject_id\")\n",
    "#take collumns\n",
    "france_large_get_ID <- france_large_ID%>% select(Run)\n",
    "\n",
    "write_tsv(france_large_get_ID, \"france_large_adenoma_get_ID.tsv\")\n",
    "\n",
    "#take ID from normal - France\n",
    "france_normal_ID <- inner_join(study1_france_normal, Study_1_metadata_France_colon_rectar, by = \"host_subject_id\")\n",
    "#take collumns\n",
    "france_normal_get_ID <- france_normal_ID%>% select(Run)\n",
    "\n",
    "write_tsv(france_normal_get_ID, \"france_normal_get_ID.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5a08d531-1d83-4190-91be-8c2ca3ce29e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mOne or more parsing issues, call `problems()` on your data frame for details, e.g.:\n",
      "  dat <- vroom(...)\n",
      "  problems(dat)”\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m128\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m57\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m  (39): Run, Assay Type, BioProject, BioSample, Center Name, config, Cons...\n",
      "\u001b[32mdbl\u001b[39m   (7): AvgSpotLen, Bases, Bytes, Collection_Date, region), geographic_lo...\n",
      "\u001b[33mlgl\u001b[39m   (1): scientific_name\n",
      "\u001b[34mdttm\u001b[39m  (6): INSDC_center_name, INSDC_first_public, project_name, INSDC_center...\n",
      "\u001b[34mdate\u001b[39m  (4): ENA-FIRST-PUBLIC (run), ENA-LAST-UPDATE (run), common_name, ENA_f...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 57</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>Bytes</th><th scope=col>Center Name</th><th scope=col>Collection_Date</th><th scope=col>config</th><th scope=col>⋯</th><th scope=col>Submitter_Id</th><th scope=col>common_name</th><th scope=col>ENA_first_public</th><th scope=col>ENA_last_update</th><th scope=col>INSDC_center_alias</th><th scope=col>create_date</th><th scope=col>version</th><th scope=col>ENA-FIRST-PUBLIC</th><th scope=col>ENA-LAST-UPDATE</th><th scope=col>scientific_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ERR1018185</td><td>WGS</td><td>190</td><td>5782402056</td><td>PRJEB10878</td><td>SAMEA3541469</td><td>5088184451</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>NA            </td><td>NA</td><td>NA</td><td>NA </td><td>2015-12-07 20:59:00</td><td>1</td><td>2015-10-01 17:02:43</td><td>2018-03-09 13:31:07</td><td>human gut metagenome</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR1018186</td><td>WGS</td><td>187</td><td>4958196559</td><td>PRJEB10878</td><td>SAMEA3541470</td><td>4469985152</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>SZAXPI003409-8</td><td>2015-10-01</td><td>2016-10-21</td><td>BGI</td><td>2015-12-07 20:48:00</td><td>1</td><td>NA</td><td>NA</td><td>NA                  </td><td>NA</td></tr>\n",
       "\t<tr><td>ERR1018187</td><td>WGS</td><td>189</td><td>5674034861</td><td>PRJEB10878</td><td>SAMEA3541471</td><td>5019710808</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>SZAXPI003409-8</td><td>2015-10-01</td><td>2016-10-21</td><td>BGI</td><td>2015-12-07 20:56:00</td><td>1</td><td>NA</td><td>NA</td><td>NA                  </td><td>NA</td></tr>\n",
       "\t<tr><td>ERR1018188</td><td>WGS</td><td>191</td><td>5343486861</td><td>PRJEB10878</td><td>SAMEA3541472</td><td>4702170554</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>SZAXPI003409-8</td><td>2015-10-01</td><td>2016-10-21</td><td>BGI</td><td>2015-12-07 20:50:00</td><td>1</td><td>NA</td><td>NA</td><td>NA                  </td><td>NA</td></tr>\n",
       "\t<tr><td>ERR1018189</td><td>WGS</td><td>189</td><td>6090719237</td><td>PRJEB10878</td><td>SAMEA3541473</td><td>5440017629</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>SZAXPI003409-8</td><td>2015-10-01</td><td>2016-10-21</td><td>BGI</td><td>2015-12-07 20:48:00</td><td>1</td><td>NA</td><td>NA</td><td>NA                  </td><td>NA</td></tr>\n",
       "\t<tr><td>ERR1018190</td><td>WGS</td><td>190</td><td>4952005733</td><td>PRJEB10878</td><td>SAMEA3541474</td><td>4415046946</td><td>BGI</td><td>2012</td><td>case</td><td>⋯</td><td>SZAXPI003409-8</td><td>2015-10-01</td><td>2016-10-21</td><td>BGI</td><td>2015-12-07 20:47:00</td><td>1</td><td>NA</td><td>NA</td><td>NA                  </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 57\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & Bytes & Center Name & Collection\\_Date & config & ⋯ & Submitter\\_Id & common\\_name & ENA\\_first\\_public & ENA\\_last\\_update & INSDC\\_center\\_alias & create\\_date & version & ENA-FIRST-PUBLIC & ENA-LAST-UPDATE & scientific\\_name\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <chr> & ⋯ & <chr> & <date> & <date> & <chr> & <dttm> & <dbl> & <dttm> & <dttm> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t ERR1018185 & WGS & 190 & 5782402056 & PRJEB10878 & SAMEA3541469 & 5088184451 & BGI & 2012 & case & ⋯ & NA             & NA & NA & NA  & 2015-12-07 20:59:00 & 1 & 2015-10-01 17:02:43 & 2018-03-09 13:31:07 & human gut metagenome & NA\\\\\n",
       "\t ERR1018186 & WGS & 187 & 4958196559 & PRJEB10878 & SAMEA3541470 & 4469985152 & BGI & 2012 & case & ⋯ & SZAXPI003409-8 & 2015-10-01 & 2016-10-21 & BGI & 2015-12-07 20:48:00 & 1 & NA & NA & NA                   & NA\\\\\n",
       "\t ERR1018187 & WGS & 189 & 5674034861 & PRJEB10878 & SAMEA3541471 & 5019710808 & BGI & 2012 & case & ⋯ & SZAXPI003409-8 & 2015-10-01 & 2016-10-21 & BGI & 2015-12-07 20:56:00 & 1 & NA & NA & NA                   & NA\\\\\n",
       "\t ERR1018188 & WGS & 191 & 5343486861 & PRJEB10878 & SAMEA3541472 & 4702170554 & BGI & 2012 & case & ⋯ & SZAXPI003409-8 & 2015-10-01 & 2016-10-21 & BGI & 2015-12-07 20:50:00 & 1 & NA & NA & NA                   & NA\\\\\n",
       "\t ERR1018189 & WGS & 189 & 6090719237 & PRJEB10878 & SAMEA3541473 & 5440017629 & BGI & 2012 & case & ⋯ & SZAXPI003409-8 & 2015-10-01 & 2016-10-21 & BGI & 2015-12-07 20:48:00 & 1 & NA & NA & NA                   & NA\\\\\n",
       "\t ERR1018190 & WGS & 190 & 4952005733 & PRJEB10878 & SAMEA3541474 & 4415046946 & BGI & 2012 & case & ⋯ & SZAXPI003409-8 & 2015-10-01 & 2016-10-21 & BGI & 2015-12-07 20:47:00 & 1 & NA & NA & NA                   & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 57\n",
       "\n",
       "| Run &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;dbl&gt; | Bases &lt;dbl&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | Bytes &lt;dbl&gt; | Center Name &lt;chr&gt; | Collection_Date &lt;dbl&gt; | config &lt;chr&gt; | ⋯ ⋯ | Submitter_Id &lt;chr&gt; | common_name &lt;date&gt; | ENA_first_public &lt;date&gt; | ENA_last_update &lt;chr&gt; | INSDC_center_alias &lt;dttm&gt; | create_date &lt;dbl&gt; | version &lt;dttm&gt; | ENA-FIRST-PUBLIC &lt;dttm&gt; | ENA-LAST-UPDATE &lt;chr&gt; | scientific_name &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ERR1018185 | WGS | 190 | 5782402056 | PRJEB10878 | SAMEA3541469 | 5088184451 | BGI | 2012 | case | ⋯ | NA             | NA | NA | NA  | 2015-12-07 20:59:00 | 1 | 2015-10-01 17:02:43 | 2018-03-09 13:31:07 | human gut metagenome | NA |\n",
       "| ERR1018186 | WGS | 187 | 4958196559 | PRJEB10878 | SAMEA3541470 | 4469985152 | BGI | 2012 | case | ⋯ | SZAXPI003409-8 | 2015-10-01 | 2016-10-21 | BGI | 2015-12-07 20:48:00 | 1 | NA | NA | NA                   | NA |\n",
       "| ERR1018187 | WGS | 189 | 5674034861 | PRJEB10878 | SAMEA3541471 | 5019710808 | BGI | 2012 | case | ⋯ | SZAXPI003409-8 | 2015-10-01 | 2016-10-21 | BGI | 2015-12-07 20:56:00 | 1 | NA | NA | NA                   | NA |\n",
       "| ERR1018188 | WGS | 191 | 5343486861 | PRJEB10878 | SAMEA3541472 | 4702170554 | BGI | 2012 | case | ⋯ | SZAXPI003409-8 | 2015-10-01 | 2016-10-21 | BGI | 2015-12-07 20:50:00 | 1 | NA | NA | NA                   | NA |\n",
       "| ERR1018189 | WGS | 189 | 6090719237 | PRJEB10878 | SAMEA3541473 | 5440017629 | BGI | 2012 | case | ⋯ | SZAXPI003409-8 | 2015-10-01 | 2016-10-21 | BGI | 2015-12-07 20:48:00 | 1 | NA | NA | NA                   | NA |\n",
       "| ERR1018190 | WGS | 190 | 4952005733 | PRJEB10878 | SAMEA3541474 | 4415046946 | BGI | 2012 | case | ⋯ | SZAXPI003409-8 | 2015-10-01 | 2016-10-21 | BGI | 2015-12-07 20:47:00 | 1 | NA | NA | NA                   | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Run        Assay Type AvgSpotLen Bases      BioProject BioSample   \n",
       "1 ERR1018185 WGS        190        5782402056 PRJEB10878 SAMEA3541469\n",
       "2 ERR1018186 WGS        187        4958196559 PRJEB10878 SAMEA3541470\n",
       "3 ERR1018187 WGS        189        5674034861 PRJEB10878 SAMEA3541471\n",
       "4 ERR1018188 WGS        191        5343486861 PRJEB10878 SAMEA3541472\n",
       "5 ERR1018189 WGS        189        6090719237 PRJEB10878 SAMEA3541473\n",
       "6 ERR1018190 WGS        190        4952005733 PRJEB10878 SAMEA3541474\n",
       "  Bytes      Center Name Collection_Date config ⋯ Submitter_Id   common_name\n",
       "1 5088184451 BGI         2012            case   ⋯ NA             <NA>       \n",
       "2 4469985152 BGI         2012            case   ⋯ SZAXPI003409-8 2015-10-01 \n",
       "3 5019710808 BGI         2012            case   ⋯ SZAXPI003409-8 2015-10-01 \n",
       "4 4702170554 BGI         2012            case   ⋯ SZAXPI003409-8 2015-10-01 \n",
       "5 5440017629 BGI         2012            case   ⋯ SZAXPI003409-8 2015-10-01 \n",
       "6 4415046946 BGI         2012            case   ⋯ SZAXPI003409-8 2015-10-01 \n",
       "  ENA_first_public ENA_last_update INSDC_center_alias  create_date\n",
       "1 <NA>             NA              2015-12-07 20:59:00 1          \n",
       "2 2016-10-21       BGI             2015-12-07 20:48:00 1          \n",
       "3 2016-10-21       BGI             2015-12-07 20:56:00 1          \n",
       "4 2016-10-21       BGI             2015-12-07 20:50:00 1          \n",
       "5 2016-10-21       BGI             2015-12-07 20:48:00 1          \n",
       "6 2016-10-21       BGI             2015-12-07 20:47:00 1          \n",
       "  version             ENA-FIRST-PUBLIC    ENA-LAST-UPDATE      scientific_name\n",
       "1 2015-10-01 17:02:43 2018-03-09 13:31:07 human gut metagenome NA             \n",
       "2 <NA>                <NA>                NA                   NA             \n",
       "3 <NA>                <NA>                NA                   NA             \n",
       "4 <NA>                <NA>                NA                   NA             \n",
       "5 <NA>                <NA>                NA                   NA             \n",
       "6 <NA>                <NA>                NA                   NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. FALSE\n",
       "2. FALSE\n",
       "3. FALSE\n",
       "4. FALSE\n",
       "5. FALSE\n",
       "6. FALSE\n",
       "7. FALSE\n",
       "8. FALSE\n",
       "9. FALSE\n",
       "10. FALSE\n",
       "11. FALSE\n",
       "12. FALSE\n",
       "13. FALSE\n",
       "14. FALSE\n",
       "15. FALSE\n",
       "16. FALSE\n",
       "17. FALSE\n",
       "18. FALSE\n",
       "19. FALSE\n",
       "20. FALSE\n",
       "21. FALSE\n",
       "22. FALSE\n",
       "23. FALSE\n",
       "24. FALSE\n",
       "25. FALSE\n",
       "26. FALSE\n",
       "27. FALSE\n",
       "28. FALSE\n",
       "29. FALSE\n",
       "30. FALSE\n",
       "31. FALSE\n",
       "32. FALSE\n",
       "33. FALSE\n",
       "34. FALSE\n",
       "35. FALSE\n",
       "36. FALSE\n",
       "37. FALSE\n",
       "38. FALSE\n",
       "39. FALSE\n",
       "40. FALSE\n",
       "41. FALSE\n",
       "42. FALSE\n",
       "43. FALSE\n",
       "44. FALSE\n",
       "45. FALSE\n",
       "46. FALSE\n",
       "47. FALSE\n",
       "48. FALSE\n",
       "49. FALSE\n",
       "50. FALSE\n",
       "51. FALSE\n",
       "52. FALSE\n",
       "53. FALSE\n",
       "54. FALSE\n",
       "55. FALSE\n",
       "56. FALSE\n",
       "57. FALSE\n",
       "58. FALSE\n",
       "59. FALSE\n",
       "60. FALSE\n",
       "61. FALSE\n",
       "62. FALSE\n",
       "63. FALSE\n",
       "64. FALSE\n",
       "65. FALSE\n",
       "66. FALSE\n",
       "67. FALSE\n",
       "68. FALSE\n",
       "69. FALSE\n",
       "70. FALSE\n",
       "71. FALSE\n",
       "72. FALSE\n",
       "73. FALSE\n",
       "74. FALSE\n",
       "75. FALSE\n",
       "76. FALSE\n",
       "77. FALSE\n",
       "78. FALSE\n",
       "79. FALSE\n",
       "80. FALSE\n",
       "81. FALSE\n",
       "82. FALSE\n",
       "83. FALSE\n",
       "84. FALSE\n",
       "85. FALSE\n",
       "86. FALSE\n",
       "87. FALSE\n",
       "88. FALSE\n",
       "89. FALSE\n",
       "90. FALSE\n",
       "91. FALSE\n",
       "92. FALSE\n",
       "93. FALSE\n",
       "94. FALSE\n",
       "95. FALSE\n",
       "96. FALSE\n",
       "97. FALSE\n",
       "98. FALSE\n",
       "99. FALSE\n",
       "100. FALSE\n",
       "101. FALSE\n",
       "102. FALSE\n",
       "103. FALSE\n",
       "104. FALSE\n",
       "105. FALSE\n",
       "106. FALSE\n",
       "107. FALSE\n",
       "108. FALSE\n",
       "109. FALSE\n",
       "110. FALSE\n",
       "111. FALSE\n",
       "112. FALSE\n",
       "113. FALSE\n",
       "114. FALSE\n",
       "115. FALSE\n",
       "116. FALSE\n",
       "117. FALSE\n",
       "118. FALSE\n",
       "119. FALSE\n",
       "120. FALSE\n",
       "121. FALSE\n",
       "122. FALSE\n",
       "123. FALSE\n",
       "124. FALSE\n",
       "125. FALSE\n",
       "126. FALSE\n",
       "127. FALSE\n",
       "128. FALSE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [97] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[121] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#####################################################################################################################################\n",
    "\n",
    "## Study 2 China cancer\n",
    "\n",
    "Study_2_china_cancer <- read_csv(\"Metadata/study_2_china_cancer_metadata.csv\")\n",
    "head(Study_2_china_cancer)\n",
    "\n",
    "duplicated(Study_2_china_cancer)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51633a09-849d-4a52-afee-6605f819110b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#select specific collumns\n",
    "Study_2_china_cancer_select <- Study_2_china_cancer[, c(\"Run\", \"config\")]\n",
    "\n",
    "#filter  by condition - case\n",
    "Study_2_china_cancer_select_case <- Study_2_china_cancer_select[Study_2_china_cancer_select$config == \"case\", ]\n",
    "\n",
    "#take collumns\n",
    "Study_2_china_cancer_case_Run <- Study_2_china_cancer_select_case %>% select(Run)\n",
    "\n",
    "write_tsv(Study_2_china_cancer_case_Run, \"Study_2_china_cancer_case_Run_ID.tsv\")\n",
    "\n",
    "\n",
    "#select control\n",
    "Study_2_china_cancer_select_control <- Study_2_china_cancer_select[Study_2_china_cancer_select$config == \"control\", ]\n",
    "\n",
    "duplicated(Study_2_china_cancer_select_control) #there is one more sample than it's enunciated in the manuscript, check if there is duplicated run or something like\n",
    "#take collumns\n",
    "Study_2_china_cancer_control_Run <- Study_2_china_cancer_select_control %>% select(Run)\n",
    "\n",
    "write_tsv(Study_2_china_cancer_control_Run, \"Study_2_china_cancer_control_Run_ID.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "98214157-9fcb-4b11-83b2-8a38615b22a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m312\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m53\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m  (34): Run, Assay Type, BioProject, BioSample, Center Name, Consent, DAT...\n",
      "\u001b[32mdbl\u001b[39m   (9): AvgSpotLen, Bases, Bytes, Collection_Date, geographic_location_(l...\n",
      "\u001b[34mdttm\u001b[39m  (6): INSDC_first_public, INSDC_last_update, ReleaseDate, create_date, ...\n",
      "\u001b[34mdate\u001b[39m  (4): ENA-FIRST-PUBLIC (run), ENA-LAST-UPDATE (run), ENA_first_public, ...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 53</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>Bytes</th><th scope=col>Center Name</th><th scope=col>Collection_Date</th><th scope=col>Consent</th><th scope=col>⋯</th><th scope=col>SRA Study</th><th scope=col>Submitter_Id</th><th scope=col>create_date</th><th scope=col>version</th><th scope=col>ENA_first_public</th><th scope=col>ENA_last_update</th><th scope=col>INSDC_center_alias</th><th scope=col>ENA-FIRST-PUBLIC</th><th scope=col>ENA-LAST-UPDATE</th><th scope=col>scientific_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ERR688358</td><td>WGS</td><td>79</td><td> 62474530</td><td>PRJEB7774</td><td>SAMEA3136748</td><td> 67759506</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31004</td><td>2015-03-09 23:28:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR688359</td><td>WGS</td><td>88</td><td>102874412</td><td>PRJEB7774</td><td>SAMEA3136648</td><td>100769689</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31009</td><td>2015-03-09 23:28:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR688360</td><td>WGS</td><td>88</td><td> 98141619</td><td>PRJEB7774</td><td>SAMEA3136638</td><td> 94869369</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31021</td><td>2015-03-09 23:30:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR688361</td><td>WGS</td><td>88</td><td> 71112063</td><td>PRJEB7774</td><td>SAMEA3136703</td><td> 69056250</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31030</td><td>2015-03-09 23:28:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR688362</td><td>WGS</td><td>89</td><td> 82353944</td><td>PRJEB7774</td><td>SAMEA3136670</td><td> 79486250</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31071</td><td>2015-03-09 23:28:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR688363</td><td>WGS</td><td>90</td><td>118458156</td><td>PRJEB7774</td><td>SAMEA3136657</td><td>112830607</td><td>BGI</td><td>2013</td><td>public</td><td>⋯</td><td>ERP008729</td><td>31112</td><td>2015-03-09 23:28:00</td><td>1</td><td>2015-03-05</td><td>2016-10-21</td><td>BGI</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 53\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & Bytes & Center Name & Collection\\_Date & Consent & ⋯ & SRA Study & Submitter\\_Id & create\\_date & version & ENA\\_first\\_public & ENA\\_last\\_update & INSDC\\_center\\_alias & ENA-FIRST-PUBLIC & ENA-LAST-UPDATE & scientific\\_name\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <chr> & ⋯ & <chr> & <dbl> & <dttm> & <dbl> & <date> & <date> & <chr> & <dttm> & <dttm> & <chr>\\\\\n",
       "\\hline\n",
       "\t ERR688358 & WGS & 79 &  62474530 & PRJEB7774 & SAMEA3136748 &  67759506 & BGI & 2013 & public & ⋯ & ERP008729 & 31004 & 2015-03-09 23:28:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\t ERR688359 & WGS & 88 & 102874412 & PRJEB7774 & SAMEA3136648 & 100769689 & BGI & 2013 & public & ⋯ & ERP008729 & 31009 & 2015-03-09 23:28:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\t ERR688360 & WGS & 88 &  98141619 & PRJEB7774 & SAMEA3136638 &  94869369 & BGI & 2013 & public & ⋯ & ERP008729 & 31021 & 2015-03-09 23:30:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\t ERR688361 & WGS & 88 &  71112063 & PRJEB7774 & SAMEA3136703 &  69056250 & BGI & 2013 & public & ⋯ & ERP008729 & 31030 & 2015-03-09 23:28:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\t ERR688362 & WGS & 89 &  82353944 & PRJEB7774 & SAMEA3136670 &  79486250 & BGI & 2013 & public & ⋯ & ERP008729 & 31071 & 2015-03-09 23:28:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\t ERR688363 & WGS & 90 & 118458156 & PRJEB7774 & SAMEA3136657 & 112830607 & BGI & 2013 & public & ⋯ & ERP008729 & 31112 & 2015-03-09 23:28:00 & 1 & 2015-03-05 & 2016-10-21 & BGI & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 53\n",
       "\n",
       "| Run &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;dbl&gt; | Bases &lt;dbl&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | Bytes &lt;dbl&gt; | Center Name &lt;chr&gt; | Collection_Date &lt;dbl&gt; | Consent &lt;chr&gt; | ⋯ ⋯ | SRA Study &lt;chr&gt; | Submitter_Id &lt;dbl&gt; | create_date &lt;dttm&gt; | version &lt;dbl&gt; | ENA_first_public &lt;date&gt; | ENA_last_update &lt;date&gt; | INSDC_center_alias &lt;chr&gt; | ENA-FIRST-PUBLIC &lt;dttm&gt; | ENA-LAST-UPDATE &lt;dttm&gt; | scientific_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ERR688358 | WGS | 79 |  62474530 | PRJEB7774 | SAMEA3136748 |  67759506 | BGI | 2013 | public | ⋯ | ERP008729 | 31004 | 2015-03-09 23:28:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "| ERR688359 | WGS | 88 | 102874412 | PRJEB7774 | SAMEA3136648 | 100769689 | BGI | 2013 | public | ⋯ | ERP008729 | 31009 | 2015-03-09 23:28:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "| ERR688360 | WGS | 88 |  98141619 | PRJEB7774 | SAMEA3136638 |  94869369 | BGI | 2013 | public | ⋯ | ERP008729 | 31021 | 2015-03-09 23:30:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "| ERR688361 | WGS | 88 |  71112063 | PRJEB7774 | SAMEA3136703 |  69056250 | BGI | 2013 | public | ⋯ | ERP008729 | 31030 | 2015-03-09 23:28:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "| ERR688362 | WGS | 89 |  82353944 | PRJEB7774 | SAMEA3136670 |  79486250 | BGI | 2013 | public | ⋯ | ERP008729 | 31071 | 2015-03-09 23:28:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "| ERR688363 | WGS | 90 | 118458156 | PRJEB7774 | SAMEA3136657 | 112830607 | BGI | 2013 | public | ⋯ | ERP008729 | 31112 | 2015-03-09 23:28:00 | 1 | 2015-03-05 | 2016-10-21 | BGI | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       Assay Type AvgSpotLen Bases     BioProject BioSample    Bytes    \n",
       "1 ERR688358 WGS        79          62474530 PRJEB7774  SAMEA3136748  67759506\n",
       "2 ERR688359 WGS        88         102874412 PRJEB7774  SAMEA3136648 100769689\n",
       "3 ERR688360 WGS        88          98141619 PRJEB7774  SAMEA3136638  94869369\n",
       "4 ERR688361 WGS        88          71112063 PRJEB7774  SAMEA3136703  69056250\n",
       "5 ERR688362 WGS        89          82353944 PRJEB7774  SAMEA3136670  79486250\n",
       "6 ERR688363 WGS        90         118458156 PRJEB7774  SAMEA3136657 112830607\n",
       "  Center Name Collection_Date Consent ⋯ SRA Study Submitter_Id\n",
       "1 BGI         2013            public  ⋯ ERP008729 31004       \n",
       "2 BGI         2013            public  ⋯ ERP008729 31009       \n",
       "3 BGI         2013            public  ⋯ ERP008729 31021       \n",
       "4 BGI         2013            public  ⋯ ERP008729 31030       \n",
       "5 BGI         2013            public  ⋯ ERP008729 31071       \n",
       "6 BGI         2013            public  ⋯ ERP008729 31112       \n",
       "  create_date         version ENA_first_public ENA_last_update\n",
       "1 2015-03-09 23:28:00 1       2015-03-05       2016-10-21     \n",
       "2 2015-03-09 23:28:00 1       2015-03-05       2016-10-21     \n",
       "3 2015-03-09 23:30:00 1       2015-03-05       2016-10-21     \n",
       "4 2015-03-09 23:28:00 1       2015-03-05       2016-10-21     \n",
       "5 2015-03-09 23:28:00 1       2015-03-05       2016-10-21     \n",
       "6 2015-03-09 23:28:00 1       2015-03-05       2016-10-21     \n",
       "  INSDC_center_alias ENA-FIRST-PUBLIC ENA-LAST-UPDATE scientific_name\n",
       "1 BGI                <NA>             <NA>            NA             \n",
       "2 BGI                <NA>             <NA>            NA             \n",
       "3 BGI                <NA>             <NA>            NA             \n",
       "4 BGI                <NA>             <NA>            NA             \n",
       "5 BGI                <NA>             <NA>            NA             \n",
       "6 BGI                <NA>             <NA>            NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "############################ Study3 - colon cancer\n",
    "\n",
    "\n",
    "Study_3_colon_cancer_metadata <- read_csv(\"Metadata/Study_3_colon_cancer_metadata.csv\")\n",
    "head(Study_3_colon_cancer_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a32f1070-c12f-413e-9cba-c1fdaf1080e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `g/week` -> `g/week...35`\n",
      "\u001b[36m•\u001b[39m `g/week` -> `g/week...36`\n",
      "\u001b[36m•\u001b[39m `g/week` -> `g/week...37`\n",
      "\u001b[36m•\u001b[39m `g/week` -> `g/week...38`\n",
      "\u001b[36m•\u001b[39m `` -> `...48`\n",
      "\u001b[36m•\u001b[39m `` -> `...50`\n",
      "\u001b[36m•\u001b[39m `` -> `...52`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 52</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample ID</th><th scope=col>State</th><th scope=col>Age (yrs)</th><th scope=col>BMI</th><th scope=col>Waist (cm)</th><th scope=col>Hip (cm)</th><th scope=col>WHR (waist-hip-ratio)</th><th scope=col>Gender (1:male, 2:female)</th><th scope=col>Fatty liver in ultrasound(1:yes, 0:no)</th><th scope=col>GGT (U/L)</th><th scope=col>⋯</th><th scope=col>1:low</th><th scope=col>1:intermediate</th><th scope=col>1:high</th><th scope=col>1:physical activity available, 0:physical activity not available</th><th scope=col>Probability of carcinoma</th><th scope=col>...48</th><th scope=col>Probability of advanced adenoma</th><th scope=col>...50</th><th scope=col>Probability of advanced adenoma add age</th><th scope=col>...52</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 31766</td><td>controls</td><td>73</td><td>32</td><td>109</td><td>120</td><td>0.908333333333333</td><td>2</td><td>1</td><td>74</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.005376344</td><td>train</td><td>0.06875000</td><td>train</td><td>0.07894737</td><td>train</td></tr>\n",
       "\t<tr><td> 31537</td><td>controls</td><td>70</td><td>31</td><td>89 </td><td>110</td><td>0.809090909090909</td><td>2</td><td>0</td><td>9 </td><td>⋯</td><td>0</td><td>1</td><td>0</td><td>1</td><td>0.346153846</td><td>train</td><td>0.06989247</td><td>train</td><td>0.08426966</td><td>train</td></tr>\n",
       "\t<tr><td> 31600</td><td>controls</td><td>70</td><td>30</td><td>101</td><td>114</td><td>0.885964912280702</td><td>2</td><td>1</td><td>15</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.111731844</td><td>train</td><td>0.08556150</td><td>train</td><td>0.02747253</td><td>train</td></tr>\n",
       "\t<tr><td> 31428</td><td>controls</td><td>69</td><td>30</td><td>96 </td><td>107</td><td>0.897196261682243</td><td>2</td><td>0</td><td>24</td><td>⋯</td><td>1</td><td>0</td><td>0</td><td>1</td><td>0.473684211</td><td>train</td><td>0.28888889</td><td>train</td><td>0.34636872</td><td>train</td></tr>\n",
       "\t<tr><td>530167</td><td>controls</td><td>72</td><td>30</td><td>110</td><td>113</td><td>0.973451327433628</td><td>2</td><td>0</td><td>48</td><td>⋯</td><td>0</td><td>1</td><td>0</td><td>1</td><td>0.693467337</td><td>train</td><td>0.29585799</td><td>train</td><td>0.17112299</td><td>train</td></tr>\n",
       "\t<tr><td>530315</td><td>controls</td><td>75</td><td>31</td><td>102</td><td>113</td><td>0.902654867256637</td><td>2</td><td>1</td><td>23</td><td>⋯</td><td>1</td><td>0</td><td>0</td><td>1</td><td>0.333333333</td><td>train</td><td>0.50990099</td><td>train</td><td>0.48979592</td><td>train</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 52\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Sample ID & State & Age (yrs) & BMI & Waist (cm) & Hip (cm) & WHR (waist-hip-ratio) & Gender (1:male, 2:female) & Fatty liver in ultrasound(1:yes, 0:no) & GGT (U/L) & ⋯ & 1:low & 1:intermediate & 1:high & 1:physical activity available, 0:physical activity not available & Probability of carcinoma & ...48 & Probability of advanced adenoma & ...50 & Probability of advanced adenoma add age & ...52\\\\\n",
       " <dbl> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  31766 & controls & 73 & 32 & 109 & 120 & 0.908333333333333 & 2 & 1 & 74 & ⋯ & 0 & 0 & 0 & 0 & 0.005376344 & train & 0.06875000 & train & 0.07894737 & train\\\\\n",
       "\t  31537 & controls & 70 & 31 & 89  & 110 & 0.809090909090909 & 2 & 0 & 9  & ⋯ & 0 & 1 & 0 & 1 & 0.346153846 & train & 0.06989247 & train & 0.08426966 & train\\\\\n",
       "\t  31600 & controls & 70 & 30 & 101 & 114 & 0.885964912280702 & 2 & 1 & 15 & ⋯ & 0 & 0 & 0 & 0 & 0.111731844 & train & 0.08556150 & train & 0.02747253 & train\\\\\n",
       "\t  31428 & controls & 69 & 30 & 96  & 107 & 0.897196261682243 & 2 & 0 & 24 & ⋯ & 1 & 0 & 0 & 1 & 0.473684211 & train & 0.28888889 & train & 0.34636872 & train\\\\\n",
       "\t 530167 & controls & 72 & 30 & 110 & 113 & 0.973451327433628 & 2 & 0 & 48 & ⋯ & 0 & 1 & 0 & 1 & 0.693467337 & train & 0.29585799 & train & 0.17112299 & train\\\\\n",
       "\t 530315 & controls & 75 & 31 & 102 & 113 & 0.902654867256637 & 2 & 1 & 23 & ⋯ & 1 & 0 & 0 & 1 & 0.333333333 & train & 0.50990099 & train & 0.48979592 & train\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 52\n",
       "\n",
       "| Sample ID &lt;dbl&gt; | State &lt;chr&gt; | Age (yrs) &lt;dbl&gt; | BMI &lt;dbl&gt; | Waist (cm) &lt;chr&gt; | Hip (cm) &lt;chr&gt; | WHR (waist-hip-ratio) &lt;chr&gt; | Gender (1:male, 2:female) &lt;dbl&gt; | Fatty liver in ultrasound(1:yes, 0:no) &lt;chr&gt; | GGT (U/L) &lt;chr&gt; | ⋯ ⋯ | 1:low &lt;chr&gt; | 1:intermediate &lt;chr&gt; | 1:high &lt;chr&gt; | 1:physical activity available, 0:physical activity not available &lt;chr&gt; | Probability of carcinoma &lt;dbl&gt; | ...48 &lt;chr&gt; | Probability of advanced adenoma &lt;dbl&gt; | ...50 &lt;chr&gt; | Probability of advanced adenoma add age &lt;dbl&gt; | ...52 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "|  31766 | controls | 73 | 32 | 109 | 120 | 0.908333333333333 | 2 | 1 | 74 | ⋯ | 0 | 0 | 0 | 0 | 0.005376344 | train | 0.06875000 | train | 0.07894737 | train |\n",
       "|  31537 | controls | 70 | 31 | 89  | 110 | 0.809090909090909 | 2 | 0 | 9  | ⋯ | 0 | 1 | 0 | 1 | 0.346153846 | train | 0.06989247 | train | 0.08426966 | train |\n",
       "|  31600 | controls | 70 | 30 | 101 | 114 | 0.885964912280702 | 2 | 1 | 15 | ⋯ | 0 | 0 | 0 | 0 | 0.111731844 | train | 0.08556150 | train | 0.02747253 | train |\n",
       "|  31428 | controls | 69 | 30 | 96  | 107 | 0.897196261682243 | 2 | 0 | 24 | ⋯ | 1 | 0 | 0 | 1 | 0.473684211 | train | 0.28888889 | train | 0.34636872 | train |\n",
       "| 530167 | controls | 72 | 30 | 110 | 113 | 0.973451327433628 | 2 | 0 | 48 | ⋯ | 0 | 1 | 0 | 1 | 0.693467337 | train | 0.29585799 | train | 0.17112299 | train |\n",
       "| 530315 | controls | 75 | 31 | 102 | 113 | 0.902654867256637 | 2 | 1 | 23 | ⋯ | 1 | 0 | 0 | 1 | 0.333333333 | train | 0.50990099 | train | 0.48979592 | train |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample ID State    Age (yrs) BMI Waist (cm) Hip (cm) WHR (waist-hip-ratio)\n",
       "1  31766    controls 73        32  109        120      0.908333333333333    \n",
       "2  31537    controls 70        31  89         110      0.809090909090909    \n",
       "3  31600    controls 70        30  101        114      0.885964912280702    \n",
       "4  31428    controls 69        30  96         107      0.897196261682243    \n",
       "5 530167    controls 72        30  110        113      0.973451327433628    \n",
       "6 530315    controls 75        31  102        113      0.902654867256637    \n",
       "  Gender (1:male, 2:female) Fatty liver in ultrasound(1:yes, 0:no) GGT (U/L) ⋯\n",
       "1 2                         1                                      74        ⋯\n",
       "2 2                         0                                      9         ⋯\n",
       "3 2                         1                                      15        ⋯\n",
       "4 2                         0                                      24        ⋯\n",
       "5 2                         0                                      48        ⋯\n",
       "6 2                         1                                      23        ⋯\n",
       "  1:low 1:intermediate 1:high\n",
       "1 0     0              0     \n",
       "2 0     1              0     \n",
       "3 0     0              0     \n",
       "4 1     0              0     \n",
       "5 0     1              0     \n",
       "6 1     0              0     \n",
       "  1:physical activity available, 0:physical activity not available\n",
       "1 0                                                               \n",
       "2 1                                                               \n",
       "3 0                                                               \n",
       "4 1                                                               \n",
       "5 1                                                               \n",
       "6 1                                                               \n",
       "  Probability of carcinoma ...48 Probability of advanced adenoma ...50\n",
       "1 0.005376344              train 0.06875000                      train\n",
       "2 0.346153846              train 0.06989247                      train\n",
       "3 0.111731844              train 0.08556150                      train\n",
       "4 0.473684211              train 0.28888889                      train\n",
       "5 0.693467337              train 0.29585799                      train\n",
       "6 0.333333333              train 0.50990099                      train\n",
       "  Probability of advanced adenoma add age ...52\n",
       "1 0.07894737                              train\n",
       "2 0.08426966                              train\n",
       "3 0.02747253                              train\n",
       "4 0.34636872                              train\n",
       "5 0.17112299                              train\n",
       "6 0.48979592                              train"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#import supplementary data to match sample IDs\n",
    "library(readxl)\n",
    "\n",
    "study3_cancer_suplem_data <-read_xlsx(\"Metadata/Study3_colon_cancer_supplementary_data.xlsx\")\n",
    "head(study3_cancer_suplem_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7e89951-ce0d-4522-8cd5-866ae9ff636c",
   "metadata": {},
   "outputs": [],
   "source": [
    "unique(study3_cancer_suplem_data$State[duplicated(study3_cancer_suplem_data$State)])\n",
    "#join data\n",
    "\n",
    "#select specific collumns\n",
    "study3_cancer_suplem_data_columm <- study3_cancer_suplem_data[, c(\"Sample ID\", \"State\",\"Gender (1:male, 2:female)\")]\n",
    "\n",
    "colnames(study3_cancer_suplem_data_columm)[1] <- \"sample_name\"\n",
    "#separate controls by male and female\n",
    "study3_cancer_suplem_data_control_male <- study3_cancer_suplem_data_columm%>%filter(State==\"controls\" & \n",
    "                                                                                      `Gender (1:male, 2:female)` ==\"1\")\n",
    "\n",
    "study3_cancer_suplem_data_control_female <- study3_cancer_suplem_data_columm%>%filter(State==\"controls\" & \n",
    "                                                                                        `Gender (1:male, 2:female)` ==\"2\")\n",
    "\n",
    "\n",
    "#separate advanced adenoma by male and female\n",
    "\n",
    "study3_cancer_suplem_data_adenoma_male <- study3_cancer_suplem_data_columm%>%filter(State==\"advanced adenoma\" & \n",
    "                                                                                        `Gender (1:male, 2:female)` ==\"1\")\n",
    "\n",
    "\n",
    "study3_cancer_suplem_data_adenoma_female <- study3_cancer_suplem_data_columm%>%filter(State==\"advanced adenoma\" & \n",
    "                                                                                      `Gender (1:male, 2:female)` ==\"2\")\n",
    "\n",
    "#separate carcinoma by male and female\n",
    "\n",
    "study3_cancer_suplem_data_carcinoma_male <- study3_cancer_suplem_data_columm%>%filter(State==\"carcinoma\" & \n",
    "                                                                                        `Gender (1:male, 2:female)` ==\"1\")\n",
    "\n",
    "study3_cancer_suplem_data_carcinoma_female <- study3_cancer_suplem_data_columm%>%filter(State==\"carcinoma\" & \n",
    "                                                                                        `Gender (1:male, 2:female)` ==\"2\")\n",
    "\n",
    "\n",
    "#match files in control\n",
    "control_male_study3 <- inner_join(study3_cancer_suplem_data_control_male,\n",
    "                                  Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "control_male_study3_run <- control_male_study3[,4]\n",
    "\n",
    "write_tsv(control_male_study3_run, \"control_male_study3.tsv\")\n",
    "\n",
    "#female\n",
    "\n",
    "control_female_study3 <- inner_join(study3_cancer_suplem_data_control_female,\n",
    "                                  Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "control_female_study3_run <- control_female_study3[,4]\n",
    "write_tsv(control_female_study3_run,\"control_female_study3.tsv\")\n",
    "\n",
    "#match files in advanced adenoma\n",
    "\n",
    "#male\n",
    "adenoma_male_study3 <- inner_join(study3_cancer_suplem_data_adenoma_male,\n",
    "                                  Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "#male Run columm\n",
    "adenoma_male_study3_run <- adenoma_male_study3[,4]\n",
    "write_tsv(adenoma_male_study3_run, \"adenoma_male_study3.tsv\")\n",
    "\n",
    "#female\n",
    "adenoma_female_study3 <- inner_join(study3_cancer_suplem_data_adenoma_female,\n",
    "                                  Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "adenoma_female_study3_run <- adenoma_female_study3[,4]\n",
    "write_tsv(adenoma_female_study3_run, \"adenoma_female_study3.tsv\")\n",
    "\n",
    "\n",
    "#match files in carcinoma\n",
    "#male\n",
    "carcinoma_male_study3 <- inner_join(study3_cancer_suplem_data_carcinoma_male,\n",
    "                                  Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "carcinoma_male_study3_run <- carcinoma_male_study3[,4]\n",
    "write_tsv(carcinoma_male_study3_run, \"carcinoma_male_study3.tsv\")\n",
    "\n",
    "\n",
    "\n",
    "#female\n",
    "carcinoma_female_study3 <- inner_join(study3_cancer_suplem_data_carcinoma_female,\n",
    "                                    Study_3_colon_cancer_metadata, by = \"sample_name\")\n",
    "\n",
    "carcinoma_female_study3_run <- carcinoma_female_study3[,4]\n",
    "write_tsv(carcinoma_female_study3_run, \"carcinoma_female_study3.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d2ad0a65-f2b0-45e0-bb55-7eac8e424be6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `ena_last_update` -> `ena_last_update...44`\n",
      "\u001b[36m•\u001b[39m `ena_last_update` -> `ena_last_update...52`\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m405\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m53\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m──────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m  (36): Run, Assay Type, BioProject, BioSample, Center Name, Consent, env...\n",
      "\u001b[32mdbl\u001b[39m   (7): Collection_Date, geographic_location_(latitude), geographic_locat...\n",
      "\u001b[34mdttm\u001b[39m  (6): INSDC_first_public, INSDC_last_update, ReleaseDate, create_date, ...\n",
      "\u001b[34mdate\u001b[39m  (4): ENA-FIRST-PUBLIC (run), ENA-LAST-UPDATE (run), ENA_first_public, ...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 53</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>Assay Type</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>Center Name</th><th scope=col>Collection_Date</th><th scope=col>Consent</th><th scope=col>ENA-FIRST-PUBLIC (run)</th><th scope=col>ENA-LAST-UPDATE (run)</th><th scope=col>environment_(biome)</th><th scope=col>⋯</th><th scope=col>ena_last_update...44</th><th scope=col>INSDC_center_alias</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>Bytes</th><th scope=col>create_date</th><th scope=col>version</th><th scope=col>ena_first_public</th><th scope=col>ena_last_update...52</th><th scope=col>Scientific_Name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;dttm&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ERR2017411</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142374</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>177</td><td>3161133051</td><td>2409737743</td><td>2017-12-15 16:14:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR2017412</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142373</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>157</td><td>2808375061</td><td>2461690116</td><td>2017-12-15 16:16:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR2017413</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142372</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>158</td><td>2807517536</td><td>2453740943</td><td>2017-12-15 16:18:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR2017414</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142371</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>158</td><td>3141515009</td><td>2756007878</td><td>2017-12-15 16:20:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR2017415</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142370</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>158</td><td>3671068081</td><td>3213635111</td><td>2017-12-15 16:23:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ERR2017416</td><td>WGS</td><td>PRJEB21528</td><td>SAMEA104142369</td><td>BEIJING GENOME INSTITUTE</td><td>2011</td><td>public</td><td>2017-12-14</td><td>2018-11-16</td><td>human gut</td><td>⋯</td><td>2017-06-27</td><td>BGI</td><td>156</td><td>3698292976</td><td>3284925472</td><td>2017-12-15 16:27:00</td><td>1</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 53\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & Assay Type & BioProject & BioSample & Center Name & Collection\\_Date & Consent & ENA-FIRST-PUBLIC (run) & ENA-LAST-UPDATE (run) & environment\\_(biome) & ⋯ & ena\\_last\\_update...44 & INSDC\\_center\\_alias & AvgSpotLen & Bases & Bytes & create\\_date & version & ena\\_first\\_public & ena\\_last\\_update...52 & Scientific\\_Name\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <date> & <date> & <chr> & ⋯ & <date> & <chr> & <dbl> & <dbl> & <dbl> & <dttm> & <dbl> & <dttm> & <dttm> & <chr>\\\\\n",
       "\\hline\n",
       "\t ERR2017411 & WGS & PRJEB21528 & SAMEA104142374 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 177 & 3161133051 & 2409737743 & 2017-12-15 16:14:00 & 1 & NA & NA & NA\\\\\n",
       "\t ERR2017412 & WGS & PRJEB21528 & SAMEA104142373 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 157 & 2808375061 & 2461690116 & 2017-12-15 16:16:00 & 1 & NA & NA & NA\\\\\n",
       "\t ERR2017413 & WGS & PRJEB21528 & SAMEA104142372 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 158 & 2807517536 & 2453740943 & 2017-12-15 16:18:00 & 1 & NA & NA & NA\\\\\n",
       "\t ERR2017414 & WGS & PRJEB21528 & SAMEA104142371 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 158 & 3141515009 & 2756007878 & 2017-12-15 16:20:00 & 1 & NA & NA & NA\\\\\n",
       "\t ERR2017415 & WGS & PRJEB21528 & SAMEA104142370 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 158 & 3671068081 & 3213635111 & 2017-12-15 16:23:00 & 1 & NA & NA & NA\\\\\n",
       "\t ERR2017416 & WGS & PRJEB21528 & SAMEA104142369 & BEIJING GENOME INSTITUTE & 2011 & public & 2017-12-14 & 2018-11-16 & human gut & ⋯ & 2017-06-27 & BGI & 156 & 3698292976 & 3284925472 & 2017-12-15 16:27:00 & 1 & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 53\n",
       "\n",
       "| Run &lt;chr&gt; | Assay Type &lt;chr&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | Center Name &lt;chr&gt; | Collection_Date &lt;dbl&gt; | Consent &lt;chr&gt; | ENA-FIRST-PUBLIC (run) &lt;date&gt; | ENA-LAST-UPDATE (run) &lt;date&gt; | environment_(biome) &lt;chr&gt; | ⋯ ⋯ | ena_last_update...44 &lt;date&gt; | INSDC_center_alias &lt;chr&gt; | AvgSpotLen &lt;dbl&gt; | Bases &lt;dbl&gt; | Bytes &lt;dbl&gt; | create_date &lt;dttm&gt; | version &lt;dbl&gt; | ena_first_public &lt;dttm&gt; | ena_last_update...52 &lt;dttm&gt; | Scientific_Name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ERR2017411 | WGS | PRJEB21528 | SAMEA104142374 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 177 | 3161133051 | 2409737743 | 2017-12-15 16:14:00 | 1 | NA | NA | NA |\n",
       "| ERR2017412 | WGS | PRJEB21528 | SAMEA104142373 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 157 | 2808375061 | 2461690116 | 2017-12-15 16:16:00 | 1 | NA | NA | NA |\n",
       "| ERR2017413 | WGS | PRJEB21528 | SAMEA104142372 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 158 | 2807517536 | 2453740943 | 2017-12-15 16:18:00 | 1 | NA | NA | NA |\n",
       "| ERR2017414 | WGS | PRJEB21528 | SAMEA104142371 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 158 | 3141515009 | 2756007878 | 2017-12-15 16:20:00 | 1 | NA | NA | NA |\n",
       "| ERR2017415 | WGS | PRJEB21528 | SAMEA104142370 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 158 | 3671068081 | 3213635111 | 2017-12-15 16:23:00 | 1 | NA | NA | NA |\n",
       "| ERR2017416 | WGS | PRJEB21528 | SAMEA104142369 | BEIJING GENOME INSTITUTE | 2011 | public | 2017-12-14 | 2018-11-16 | human gut | ⋯ | 2017-06-27 | BGI | 156 | 3698292976 | 3284925472 | 2017-12-15 16:27:00 | 1 | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Run        Assay Type BioProject BioSample      Center Name             \n",
       "1 ERR2017411 WGS        PRJEB21528 SAMEA104142374 BEIJING GENOME INSTITUTE\n",
       "2 ERR2017412 WGS        PRJEB21528 SAMEA104142373 BEIJING GENOME INSTITUTE\n",
       "3 ERR2017413 WGS        PRJEB21528 SAMEA104142372 BEIJING GENOME INSTITUTE\n",
       "4 ERR2017414 WGS        PRJEB21528 SAMEA104142371 BEIJING GENOME INSTITUTE\n",
       "5 ERR2017415 WGS        PRJEB21528 SAMEA104142370 BEIJING GENOME INSTITUTE\n",
       "6 ERR2017416 WGS        PRJEB21528 SAMEA104142369 BEIJING GENOME INSTITUTE\n",
       "  Collection_Date Consent ENA-FIRST-PUBLIC (run) ENA-LAST-UPDATE (run)\n",
       "1 2011            public  2017-12-14             2018-11-16           \n",
       "2 2011            public  2017-12-14             2018-11-16           \n",
       "3 2011            public  2017-12-14             2018-11-16           \n",
       "4 2011            public  2017-12-14             2018-11-16           \n",
       "5 2011            public  2017-12-14             2018-11-16           \n",
       "6 2011            public  2017-12-14             2018-11-16           \n",
       "  environment_(biome) ⋯ ena_last_update...44 INSDC_center_alias AvgSpotLen\n",
       "1 human gut           ⋯ 2017-06-27           BGI                177       \n",
       "2 human gut           ⋯ 2017-06-27           BGI                157       \n",
       "3 human gut           ⋯ 2017-06-27           BGI                158       \n",
       "4 human gut           ⋯ 2017-06-27           BGI                158       \n",
       "5 human gut           ⋯ 2017-06-27           BGI                158       \n",
       "6 human gut           ⋯ 2017-06-27           BGI                156       \n",
       "  Bases      Bytes      create_date         version ena_first_public\n",
       "1 3161133051 2409737743 2017-12-15 16:14:00 1       <NA>            \n",
       "2 2808375061 2461690116 2017-12-15 16:16:00 1       <NA>            \n",
       "3 2807517536 2453740943 2017-12-15 16:18:00 1       <NA>            \n",
       "4 3141515009 2756007878 2017-12-15 16:20:00 1       <NA>            \n",
       "5 3671068081 3213635111 2017-12-15 16:23:00 1       <NA>            \n",
       "6 3698292976 3284925472 2017-12-15 16:27:00 1       <NA>            \n",
       "  ena_last_update...52 Scientific_Name\n",
       "1 <NA>                 NA             \n",
       "2 <NA>                 NA             \n",
       "3 <NA>                 NA             \n",
       "4 <NA>                 NA             \n",
       "5 <NA>                 NA             \n",
       "6 <NA>                 NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#################################################################################################\n",
    "# Aterosclerosis - Study 01 - Chinese cohort\n",
    "\n",
    "\n",
    "library(readr)\n",
    "atero_study01 <- read_csv(\"Metadata/atero_study01.csv\")\n",
    "head(atero_study01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "5da7e2ae-1363-4dcd-907d-48022ab617ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 55</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample ID</th><th scope=col>ACVD status</th><th scope=col>Coronary type</th><th scope=col>Age (year)</th><th scope=col>Gender</th><th scope=col>Height (cm)</th><th scope=col>Weight (kg)</th><th scope=col>Body Mass Index (BMI)</th><th scope=col>Waist (cm)</th><th scope=col>Hip (cm)</th><th scope=col>⋯</th><th scope=col>Bisoprolol_9</th><th scope=col>Fondaparinux Sodium_10</th><th scope=col>Metoprolol_19</th><th scope=col>Insulin aspart_6</th><th scope=col>Isosorbide dinitrate_10</th><th scope=col>Acarbose_7</th><th scope=col>Captopril Tablets_6</th><th scope=col>Estazolam_6</th><th scope=col>Nitroglycerin Tablets_8</th><th scope=col>Potassium chloride_32</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>N001</td><td>0</td><td>NA</td><td>52</td><td>male  </td><td>165</td><td>65  </td><td>23.88</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N002</td><td>0</td><td>NA</td><td>53</td><td>male  </td><td>160</td><td>63  </td><td>24.61</td><td>78</td><td>90</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N003</td><td>0</td><td>NA</td><td>48</td><td>male  </td><td>166</td><td>70  </td><td>25.4 </td><td>92</td><td>98</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N005</td><td>0</td><td>NA</td><td>53</td><td>female</td><td>158</td><td>57.5</td><td>23.03</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N006</td><td>0</td><td>NA</td><td>53</td><td>female</td><td>150</td><td>60  </td><td>26.67</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><td>N009</td><td>0</td><td>NA</td><td>60</td><td>male  </td><td>177</td><td>70  </td><td>22.34</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 55\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Sample ID & ACVD status & Coronary type & Age (year) & Gender & Height (cm) & Weight (kg) & Body Mass Index (BMI) & Waist (cm) & Hip (cm) & ⋯ & Bisoprolol\\_9 & Fondaparinux Sodium\\_10 & Metoprolol\\_19 & Insulin aspart\\_6 & Isosorbide dinitrate\\_10 & Acarbose\\_7 & Captopril Tablets\\_6 & Estazolam\\_6 & Nitroglycerin Tablets\\_8 & Potassium chloride\\_32\\\\\n",
       " <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t N001 & 0 & NA & 52 & male   & 165 & 65   & 23.88 & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t N002 & 0 & NA & 53 & male   & 160 & 63   & 24.61 & 78 & 90 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t N003 & 0 & NA & 48 & male   & 166 & 70   & 25.4  & 92 & 98 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t N005 & 0 & NA & 53 & female & 158 & 57.5 & 23.03 & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t N006 & 0 & NA & 53 & female & 150 & 60   & 26.67 & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t N009 & 0 & NA & 60 & male   & 177 & 70   & 22.34 & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 55\n",
       "\n",
       "| Sample ID &lt;chr&gt; | ACVD status &lt;dbl&gt; | Coronary type &lt;chr&gt; | Age (year) &lt;chr&gt; | Gender &lt;chr&gt; | Height (cm) &lt;chr&gt; | Weight (kg) &lt;chr&gt; | Body Mass Index (BMI) &lt;chr&gt; | Waist (cm) &lt;chr&gt; | Hip (cm) &lt;chr&gt; | ⋯ ⋯ | Bisoprolol_9 &lt;chr&gt; | Fondaparinux Sodium_10 &lt;chr&gt; | Metoprolol_19 &lt;chr&gt; | Insulin aspart_6 &lt;chr&gt; | Isosorbide dinitrate_10 &lt;chr&gt; | Acarbose_7 &lt;chr&gt; | Captopril Tablets_6 &lt;chr&gt; | Estazolam_6 &lt;chr&gt; | Nitroglycerin Tablets_8 &lt;chr&gt; | Potassium chloride_32 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| N001 | 0 | NA | 52 | male   | 165 | 65   | 23.88 | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| N002 | 0 | NA | 53 | male   | 160 | 63   | 24.61 | 78 | 90 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| N003 | 0 | NA | 48 | male   | 166 | 70   | 25.4  | 92 | 98 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| N005 | 0 | NA | 53 | female | 158 | 57.5 | 23.03 | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| N006 | 0 | NA | 53 | female | 150 | 60   | 26.67 | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| N009 | 0 | NA | 60 | male   | 177 | 70   | 22.34 | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample ID ACVD status Coronary type Age (year) Gender Height (cm) Weight (kg)\n",
       "1 N001      0           NA            52         male   165         65         \n",
       "2 N002      0           NA            53         male   160         63         \n",
       "3 N003      0           NA            48         male   166         70         \n",
       "4 N005      0           NA            53         female 158         57.5       \n",
       "5 N006      0           NA            53         female 150         60         \n",
       "6 N009      0           NA            60         male   177         70         \n",
       "  Body Mass Index (BMI) Waist (cm) Hip (cm) ⋯ Bisoprolol_9\n",
       "1 23.88                 NA         NA       ⋯ NA          \n",
       "2 24.61                 78         90       ⋯ NA          \n",
       "3 25.4                  92         98       ⋯ NA          \n",
       "4 23.03                 NA         NA       ⋯ NA          \n",
       "5 26.67                 NA         NA       ⋯ NA          \n",
       "6 22.34                 NA         NA       ⋯ NA          \n",
       "  Fondaparinux Sodium_10 Metoprolol_19 Insulin aspart_6 Isosorbide dinitrate_10\n",
       "1 NA                     NA            NA               NA                     \n",
       "2 NA                     NA            NA               NA                     \n",
       "3 NA                     NA            NA               NA                     \n",
       "4 NA                     NA            NA               NA                     \n",
       "5 NA                     NA            NA               NA                     \n",
       "6 NA                     NA            NA               NA                     \n",
       "  Acarbose_7 Captopril Tablets_6 Estazolam_6 Nitroglycerin Tablets_8\n",
       "1 NA         NA                  NA          NA                     \n",
       "2 NA         NA                  NA          NA                     \n",
       "3 NA         NA                  NA          NA                     \n",
       "4 NA         NA                  NA          NA                     \n",
       "5 NA         NA                  NA          NA                     \n",
       "6 NA         NA                  NA          NA                     \n",
       "  Potassium chloride_32\n",
       "1 NA                   \n",
       "2 NA                   \n",
       "3 NA                   \n",
       "4 NA                   \n",
       "5 NA                   \n",
       "6 NA                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "atero_supple_data <- read_xlsx(\"Metadata/atero_study01.xlsx\", skip = 2, col_names = TRUE)\n",
    "head(atero_supple_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17f3437f-0aa3-4f34-b6ea-5c5f4a01d9d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#control\n",
    "\n",
    "#male\n",
    "atero_male_control_suplem <- atero_supple_data%>%filter(str_detect(`Sample ID`, \"N\") & \n",
    "                                                          Gender ==\"male\")\n",
    "\n",
    "#separate column and match with main metadata to get Run ID\n",
    "\n",
    "atero_male_control_suplem_ID <- atero_male_control_suplem[,1]\n",
    "colnames(atero_male_control_suplem_ID) <- \"Sample_name\"\n",
    "\n",
    "atero_male_final_ID <- inner_join(atero_male_control_suplem_ID,atero_study01, by= \"Sample_name\")\n",
    "\n",
    "#separate desided column with ID run\n",
    "atero_male_control_ID <- atero_male_final_ID[,2]\n",
    "write_tsv(atero_male_control_ID, \"atero_study1_male_control_ID\")\n",
    "\n",
    "#female\n",
    "atero_female_control_suplem <- atero_supple_data%>%filter(str_detect(`Sample ID`, \"N\") & \n",
    "                                                          Gender ==\"female\")\n",
    "\n",
    "#separate column and match with main metadata to get Run ID\n",
    "atero_female_control_suplem_ID <- atero_female_control_suplem[,1]\n",
    "colnames(atero_female_control_suplem_ID) <- \"Sample_name\"\n",
    "\n",
    "atero_female_final_ID <- inner_join(atero_female_control_suplem_ID,atero_study01, by= \"Sample_name\")\n",
    "\n",
    "#separate desided column with ID run\n",
    "atero_female_control_ID <- atero_female_final_ID[,2]\n",
    "write_tsv(atero_female_control_ID, \"atero_study1_female_control_ID\")\n",
    "\n",
    "\n",
    "#disease ID\n",
    "\n",
    "#male\n",
    "atero_male_disease_suplem <- atero_supple_data%>%filter(str_detect(`Sample ID`, \"ZSL\") & \n",
    "                                                          Gender ==\"male\")\n",
    "\n",
    "#separate column and match with main metadata to get Run ID\n",
    "\n",
    "atero_male_disease_suplem_ID <- atero_male_disease_suplem[,1]\n",
    "colnames(atero_male_disease_suplem_ID) <- \"Sample_name\"\n",
    "\n",
    "atero_disease_male_final_ID <- inner_join(atero_male_disease_suplem_ID,atero_study01, by= \"Sample_name\")\n",
    "\n",
    "#separate desided column with ID run\n",
    "atero_male_disease_ID <- atero_disease_male_final_ID[,2]\n",
    "write_tsv(atero_male_disease_ID, \"atero_study1_male_disease_ID\")\n",
    "\n",
    "\n",
    "#female\n",
    "atero_female_disease_suplem <- atero_supple_data%>%filter(str_detect(`Sample ID`, \"ZSL\") & \n",
    "                                                          Gender ==\"female\")\n",
    "\n",
    "#separate column and match with main metadata to get Run ID\n",
    "\n",
    "atero_female_disease_suplem_ID <- atero_female_disease_suplem[,1]\n",
    "colnames(atero_female_disease_suplem_ID) <- \"Sample_name\"\n",
    "\n",
    "atero_disease_female_final_ID <- inner_join(atero_female_disease_suplem_ID,atero_study01, by= \"Sample_name\")\n",
    "\n",
    "#separate desided column with ID run\n",
    "atero_female_disease_ID <- atero_disease_female_final_ID[,2]\n",
    "write_tsv(atero_female_disease_ID, \"atero_study1_female_disease_ID\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
